upstairs/src/downstairs.rs

```diff
@@ -3673,6 +3841,551 @@ impl Downstairs {
+                            session_id,
+                            repairs: repairs.clone(),
+                            aborted,
+                        },
+                    )
+                    .await
+            })
+            .await
+            {
+                Ok(_) => {
+                    info!(log, "notified Nexus of reconcile finish");
+                }
+
+                Err(e) => {
+                    error!(
+                        log,
+                        "failed to notify Nexus of reconcile finish! {e}"
+                    );
+                }
+            }
+        });
+    }
+
+    #[cfg(feature = "notify-nexus")]
+    fn notify_nexus_of_reconcile_progress(
+        &self,
+        reconcile_id: Uuid,
+        current_task: usize,
+        task_count: usize,
+    ) {
+        let upstairs_id: TypedUuid<UpstairsKind> =
+            TypedUuid::from_untyped_uuid(self.cfg.upstairs_id);
+        let repair_id: TypedUuid<UpstairsRepairKind> =
+            TypedUuid::from_untyped_uuid(reconcile_id);
+
+        let now = Utc::now();
+        let log = self.log.new(o!("reconcile" => repair_id.to_string()));
+
+        // Spawn a task so we don't block the main loop talking to
+        // Nexus.
+        let target_addrs = self.get_target_addrs();
+        tokio::spawn(async move {
+            let Some(nexus_client) =
+                get_nexus_client(&log, &target_addrs).await
+            else {
+                // Exit if no Nexus client returned from DNS - our notification
+                // is best effort.
+                error!(
+                    log,
+                    "no Nexus client from DNS, aborting progress notification"
+                );
+                return;
+            };
+
+            match omicron_common::retry_until_known_result(&log, || async {
+                nexus_client
+                    .cpapi_upstairs_repair_progress(
+                        &upstairs_id,
+                        &repair_id,
+                        &RepairProgress {
+                            time: now,
+                            // surely we won't have usize::MAX extents
+                            current_item: current_task as i64,
+                            // i am serious, and don't call me shirley
+                            total_items: task_count as i64,
+                        },
+                    )
+                    .await
+            })
+            .await
+            {
+                Ok(_) => {
+                    info!(log, "notified Nexus of reconcile progress");
+                }
+
+                Err(e) => {
+                    error!(
+                        log,
+                        "failed to notify Nexus of reconcile progress! {e}"
+                    );
+                }
+            }
+        });
+    }
+
+    #[cfg(feature = "notify-nexus")]
+    pub(crate) fn notify_nexus_of_client_task_stopped(
```

Comment:
I'm betting that this will be too noisy.
Lets see though.  And really, we don't need a "still not connected" kind of status.
If things work well, we just need an "Offline", then an "I'm back".

But, I think the client task stops and starts a bunch while trying to reconnect.